packages/realm-react/src/index.tsx

```diff
@@ -135,11 +54,51 @@ type RealmContext = {
  * @param realmConfig - {@link Realm.Configuration} used to open the Realm
  * @returns An object containing a `RealmProvider` component, and `useRealm`, `useQuery` and `useObject` hooks
  */
-export const createRealmContext: (realmConfig?: Realm.Configuration) => RealmContext = (
-  realmConfig: Realm.Configuration = {},
-) => {
-  const RealmContext = createContext<Realm | null>(null);
-  const RealmProvider = createRealmProvider(realmConfig, RealmContext);
+export function createRealmContext(realmConfig: Realm.Configuration): RealmContext<RealmProviderFromConfigFC>;
+/**
+ * Creates Realm React hooks and Provider component for a given Realm instance.
+ * @example
+ * ```
+ * const realm = await Realm.open({ path: "example.realm" });
+ * const {useRealm, useQuery, useObject, RealmProvider} = createRealmContext(realm);
+ * ```
+ * @param realm - {@link Realm} instance
+ * @returns An object containing a `RealmProvider` component, and `useRealm`, `useQuery` and `useObject` hooks
+ */
+export function createRealmContext(realm: Realm): RealmContext<RealmProviderFromRealmInstanceFC>;
+/**
+ * Creates Realm React hooks and Provider component.
+ * @example
+ * ```
+ * class Task extends Realm.Object {
+ *  ...
+ *  static schema = {
+ *    name: 'Task',
+ *    primaryKey: '_id',
+ *    properties: {
+ *      ...
+ *    },
+ *  };
+ * }
+ * const {useRealm, useQuery, useObject, RealmProvider} = createRealmContext();
+ * ...
+ * <RealmProvider schema={[Task]}></RealmProvider>
+ * ```
+ *  @example
+ * ```
+ * const realm = await Realm.open({ path: "example.realm" });
```

Comment:
I agree with the other place since we had the schema already defined and it's nice to show equivalence to the config-based provider but I feel like for purpose of this manually managed realm-centric bit it's not really necessary to mention a schema.

Reply:
```ts
const realm = new Realm({ schema: [...] });
```
I can do this instead